Content.Shared/Utility/ContentHelpers.cs

```diff
@@ -56,5 +56,53 @@ public static int RoundToLevels(double actual, double max, int levels)
                 return (int)Math.Floor(preround);
             }
         }
+
+        /// <summary>
+        /// Returns the segment <paramref name="actual"/> lies on on a decimal scale from 0 to <paramref name="max"/> divided into
+        /// <paramref name="levels"/> sections. In less mathematical terms, same as <see cref="RoundToLevels"/>
+        /// except <paramref name="actual"/> is rounded to the nearest matching level instead of 0 and the highest level being
+        /// precisely 0 and max and no other value.
+        /// </summary>
+        /// <example>
+        /// You have a 5-segment progress bar used to display a percentile value.
+        /// You want the display to match the percentile value as accurately as possible, so that eg.
+        /// 95% is rounded up to 5, 89.99% is rounded down to 4, 15% is rounded up to 1 and 5% is rounded down
+        /// to 0, in terms of number of segments lit.
+        /// In this case you would use <code>RoundToNearestLevels(value, max, 5)</code>
+        /// </example>
+        /// <param name="actual">The point to be rounded to the nearest level.</param>
+        /// <param name="max">The maximum value of the scale.</param>
+        /// <param name="levels">Number of segments the scale is subdivided into.</param>
+        /// <returns>The segment <paramref name="actual"/> lies on.</returns>
+        /// <exception cref="ArgumentException"></exception>
+        public static int RoundToNearestLevels(double actual, double max, int levels)
```

Comment:
Because of how the cell recharging and discharging works wrt. floating point errors, it might never actually reach max or min. Especially min, because most devices stop using power when there's less than 1 second of charge remaining.

Reply:
Max should be fine in chargers based on what I saw as it will clamp it.

As for the minimum some stuff will use all of a charge (e.g. hitscan) and some stuff requires a minimum (stunbatons).

Reply:
Either way with only 5 sprite states to go around having only 3 intermediates is too few IMO.